### PR TITLE
Balancer: Fee collector test deployment

### PR DESCRIPTION
**Pull request by facuspagnuolo**
This PR prepares a Balancer fee collector SV deployment for Mainnet<>Optimism only for testing purposes. However, all the logic has been implemented. Once we test successfully some use cases, we can move forward and update the deployment inputs to deploy the production instance.

### CODE REVIEW

File: tasks/2023032101-balancer-fee-collector-l1/input.mainnet.ts [hunk 1]
```diff
@@ -0,0 +1,131 @@
...
+      priceFeedParams: [
+        { base: tokens.DAI, quote: tokens.WETH, feed: chainlink.DAI_ETH },
+        { base: tokens.USDC, quote: tokens.WETH, feed: chainlink.USDC_ETH },
+      ],
+      priceOracle: PriceOracle,
+      swapConnector: SwapConnector,
+      bridgeConnector: BridgeConnector,
+      swapFee: { pct: 0, cap: 0, token: ZERO_ADDRESS, period: 0 },
+      bridgeFee: { pct: fp(0.005), cap: 0, token: ZERO_ADDRESS, period: 0 }, // 0.5%
+      withdrawFee: { pct: 0, cap: 0, token: ZERO_ADDRESS, period: 0 },
```
Comment (dmf7z) [inline tasks/2023032101-balancer-fee-collector-l1/input.mainnet.ts:54]: We charge a withdraw fee of 2%

File: tasks/2023032101-balancer-fee-collector-l1/input.mainnet.ts [hunk 2]
```diff
@@ -0,0 +1,131 @@
...
+      performanceFee: { pct: 0, cap: 0, token: ZERO_ADDRESS, period: 0 },
+    },
+    claimerActionParams: {
+      admin: owner,
+      managers,
+      oracleSigner: BOT,
+      protocolFeeWithdrawer,
+      tokenThresholdActionParams: {
+        token: tokens.USDC,
+        amount: toUSDC(1),
```
Comment (dmf7z) [inline tasks/2023032101-balancer-fee-collector-l1/input.mainnet.ts:64]: Min 100 USDC

File: tasks/2023032101-balancer-fee-collector-l1/input.mainnet.ts [hunk 3]
```diff
@@ -0,0 +1,131 @@
...
+      managers,
+      tokenOut: tokens.USDC,
+      swapSigner: paraswapSigner,
+      deniedTokens: [], // none
+      defaultMaxSlippage: fp(0.002), // 0.2 %
+      customSlippageTokens: [], // none
+      customSlippageValues: [], // none
+      tokenThresholdActionParams: {
+        token: tokens.USDC,
+        amount: toUSDC(10),
```
Comment (dmf7z) [inline tasks/2023032101-balancer-fee-collector-l1/input.mainnet.ts:83]: 5000 USDC

File: tasks/2023032101-balancer-fee-collector-l1/input.mainnet.ts [hunk 4]
```diff
@@ -0,0 +1,131 @@
...
```
Comment (dmf7z) [inline tasks/2023032101-balancer-fee-collector-l1/input.mainnet.ts:102]: 5000 USDC

**Review by dmf7z — COMMENTED** [review]
I know we will discuss this on Monday, but wanted to leave some notes on the configuration

File: tasks/2023032102-balancer-fee-collector-l2/input.optimism.ts [hunk 5]
```diff
@@ -0,0 +1,137 @@
...
+
+const Registry = new Task('2023010602-registry-v2')
+const SmartVault = new Task('2023010603-smart-vault-v3')
+const PriceOracle = new Task('2023010604-price-oracle-v2')
+const SwapConnector = new Task('2023010605-swap-connector-v5')
+const BridgeConnector = new Task('2023010606-bridge-connector-v1')
+const SmartVaultsFactory = new Task('2023010607-smart-vaults-factory-v1')
+const MimicFeeCollector = new Task('2023010701-mimic-fee-collector-l2')
+
+const owner = '0x0000000000000000000000000000000000000001' // TODO: Balancer multisig
```
Comment (dmf7z) [inline tasks/2023032102-balancer-fee-collector-l2/input.optimism.ts:20]: We can put. the test account here 0xe649B71783d5008d10a96b6871e3840a398d4F06

File: tasks/2023032102-balancer-fee-collector-l2/input.optimism.ts [hunk 6]
```diff
@@ -0,0 +1,137 @@
...
+const mimic = OWNER_EOA
+const managers: string[] = [] // no managers
+const relayers = [BOT]
+const feeCollector = MimicFeeCollector.key('SmartVault')
+
+const paraswapSigner = '0x6278c27CF5534F07fA8f1Ab6188a155cb8750FFA'
+const protocolFeeWithdrawer = '0x0000000000000000000000000000000000000002'
```
Comment (dmf7z) [inline tasks/2023032102-balancer-fee-collector-l2/input.optimism.ts:27]: We will deploy a mock for this aftwerwards

File: tasks/2023032102-balancer-fee-collector-l2/input.optimism.ts [hunk 7]
```diff
@@ -0,0 +1,137 @@
...
+      strategies: [],
+      priceFeedParams: [
+        { base: tokens.WETH, quote: USD, feed: chainlink.ETH_USD },
+        { base: tokens.USDC, quote: USD, feed: chainlink.USDC_USD },
+        { base: tokens.DAI, quote: USD, feed: chainlink.DAI_USD },
+      ],
+      priceOracle: PriceOracle,
+      swapConnector: SwapConnector,
+      bridgeConnector: BridgeConnector,
+      swapFee: { pct: fp(0.001), cap: 0, token: ZERO_ADDRESS, period: 0 }, // 0.1 %
```
Comment (dmf7z) [inline tasks/2023032102-balancer-fee-collector-l2/input.optimism.ts:55]: no fee

File: tasks/2023032102-balancer-fee-collector-l2/input.optimism.ts [hunk 8]
```diff
@@ -0,0 +1,137 @@
...
+      bridgeFee: { pct: fp(0.002), cap: 0, token: ZERO_ADDRESS, period: 0 }, // 0.2 %
```
Comment (dmf7z) [inline tasks/2023032102-balancer-fee-collector-l2/input.optimism.ts:56]: no fee

**Review by dmf7z — COMMENTED** [review]
Some more comments for optimism

File: tasks/2023032102-balancer-fee-collector-l2/input.optimism.ts (hunk 5 above)
Comment (facuspagnuolo) [reply tasks/2023032102-balancer-fee-collector-l2/input.optimism.ts:20]: Let's use the real one `0x09Df1626110803C7b3b07085Ef1E053494155089`

File: tasks/2023032102-balancer-fee-collector-l2/input.optimism.ts (hunk 6 above)
Comment (facuspagnuolo) [reply tasks/2023032102-balancer-fee-collector-l2/input.optimism.ts:27]: Deployed at `0xaccaE4A956393Cbc99e1A00006F0F2F9cF718347`

File: tasks/2023032101-balancer-fee-collector-l1/input.mainnet.ts [hunk 9]
```diff
@@ -0,0 +1,131 @@
...
+    smartVaultParams: {
+      factory: SmartVaultsFactory,
+      impl: SmartVault,
+      admin: owner,
+      feeCollector,
+      feeCollectorAdmin: mimic,
+      strategies: [],
+      priceFeedParams: [
+        { base: tokens.DAI, quote: tokens.WETH, feed: chainlink.DAI_ETH },
+        { base: tokens.USDC, quote: tokens.WETH, feed: chainlink.USDC_ETH },
```
[the remark this answers is not in the document]
Comment (facuspagnuolo) [reply tasks/2023032101-balancer-fee-collector-l1/input.mainnet.ts:47]: Added `0xc1438aa3823a6ba0c159cfa8d98df5a994ba120b`

**Pull request merged**